packages/material-ui-styles/src/getStylesCreator/getStylesCreator.js

```diff
@@ -1,6 +1,23 @@
 import { deepmerge } from '@material-ui/utils';
 import noopTheme from './noopTheme';
 
+const capitalize = (string) => {
+  return string.charAt(0).toUpperCase() + string.slice(1)
+}
+
+// This should be defined somehwere per component
+// ideally each component would define this for itself...
+const propsToClassKey = (matcher) => {
```

Comment:
@oliviertassinari I definitely think we should expose a method like this per component

Reply:
Do you have a specific use case in mind?

Reply:
I created this method based on how the classes are applied on the Button. Are these generic enough to be used in all components? Like for example that the variant + size props are generated in the format: `textSizeSmall`, or are these rules component specifics? If each component provide a method like this, it may be safer, but maybe there is no need for it?

Reply:
> these rules component specifics

I think that we should aim for consistency, I'm not sure we always apply the same rule, I wish we did.

---

On the generation of class names side of things, I think that such a prop or something similar could help. Basically, from what I understand, we will have the following requirements:

- Keep the same support for the `classes` prop.
- Have the class name generated outside of JSS's class name generator, right inside the unstyled components.
- Provide default values for these classes that can be targeted, related to #21658.

Maybe

- Have a way to output minified class names? changing the `Mui` prefix to match your company initials?